packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/ButtonItem/buttonItem.js

```diff
@@ -0,0 +1,159 @@
+import React, { useCallback } from 'react';
+import Button from '../../../../Button/button';
+import { arrayOf, oneOf, string, bool, object } from 'prop-types';
+import { withRouter } from '@magento/venia-drivers';
+import resolveLinkProps from '../../resolveLinkProps';
+import { compose } from 'redux';
+
+/**
+ * Page Builder ButtonItem component.
+ *
+ * This component is part of the Page Builder / PWA integration. It can be consumed without Page Builder.
+ *
+ * @typedef ButtonItem
+ * @kind functional component
+ *
+ * @param {props} props React component props
+ *
+ * @returns {React.Element} A React component that displays a button.
+ */
+const ButtonItem = props => {
+    const {
+        buttonType,
+        link,
+        linkType,
+        openInNewTab = false,
+        text,
+        textAlign,
+        border,
+        borderColor,
+        borderWidth,
+        borderRadius,
+        marginTop,
+        marginRight,
+        marginBottom,
+        marginLeft,
+        paddingTop,
+        paddingRight,
+        paddingBottom,
+        paddingLeft,
+        cssClasses = []
+    } = props;
+
+    const dynamicInnerStyles = {
+        textAlign,
+        border,
+        borderColor,
+        borderWidth,
+        borderRadius,
+        marginTop,
+        marginRight,
+        marginBottom,
+        marginLeft,
+        paddingTop,
+        paddingRight,
+        paddingBottom,
+        paddingLeft
+    };
+
+    let linkProps = {};
+    let url = '';
+    if (typeof link === 'string') {
+        linkProps = resolveLinkProps(link, linkType);
+        url = (linkProps.to ? linkProps.to : linkProps.href).trim();
+    }
+
+    const typeToPriorityMapping = {
+        primary: 'high',
+        secondary: 'normal',
+        link: 'low'
+    };
+
+    const handleClick = useCallback(() => {
+        if (!url) {
+            return;
+        }
+
+        if (openInNewTab) {
+            window.open(url, '_blank');
+        } else if (linkProps.to) {
+            props.history.push(url);
+        } else {
+            window.location.assign(url);
+        }
+    }, [openInNewTab, props.history, url, linkProps.to]);
```

Comment:
I left a [comment](https://github.com/magento/pwa-studio/pull/1848/files#r333615434) on the individual PR for this component, and it went unanswered. Does `props.history` change?